### PR TITLE
Fix CI workflows; change to run against PR

### LINKED ISSUE #2: Update GitHub workflows
Once everything is public, there are 2 changes we can make to the GitHub workflows:
1. Update integration test & CD workflows to use the names of the published artifacts to run successfully.
2. Update unit & integration tests to run during PR rather than merging into `main`

### PR DESCRIPTION
**Pull request by ohltyler**
Signed-off-by: Tyler Ohlsen <ohltyler@amazon.com>

### Description

Fixes the CI workflow to run against the OpenSearch staging docker images with the `latest` tag. Also updates the workflow to run against PRs and not just on merges to main.

NOTE: this is still checking out the `1.0.0-beta1` tag for the core repos. This workflow (as well as the other repos) will be changed once there is a dedicated branch or consistent way for plugins to build and test against the latest versions of the core repos.

### Issues Resolved

Resolves #2

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/e2e-tests-workflow.yml [hunk 1]
```diff
@@ -99,20 +102,23 @@ jobs:
...
-            echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$opensearch_version" >> Dockerfile
+            echo "FROM $OPENSEARCH_DOCKER_IMAGE:latest" >> Dockerfile
             ## The OpenSearch rest test client uses http by default.
             ## Need to disable the security plugin to call the rest api over http.
-            echo "RUN if [ -d /usr/share/OpenSearch/plugins/opensearch-security ]; then /usr/share/OpenSearch/bin/OpenSearch-plugin remove opensearch-security; fi" >> Dockerfile
+            ## Also need to remove any remaining security settings specified in opensearch.yml
+            echo "RUN if [ -d /usr/share/opensearch/plugins/opensearch-security ]; then /usr/share/opensearch/bin/opensearch-plugin remove opensearch-security; fi" >> Dockerfile
+            echo "RUN sed -n -i '1,/Start OpenSearch Security Demo Configuration/p;/End OpenSearch Security Demo Configuration/,\$p'" >> Dockerfile
+            truncate -s -1 Dockerfile
+            echo -n " /usr/share/opensearch/config/opensearch.yml" >> Dockerfile
```
Comment (ohltyler) [inline .github/workflows/e2e-tests-workflow.yml:121]: Had to use a workaround to get the full `sed` command to get written to the `Dockerfile` correctly. The purpose of the `sed` command is to remove the extra added security settings in the demo `opensearch.yml` file. Without removing, the cluster has errors, since no plugin is able to recognize the security-related settings (and we are purposefully removing the security plugin in this test case).

Comment (ylwu-amzn) [reply .github/workflows/e2e-tests-workflow.yml:121]: > no plugin is able to recognize the security-related settings

Is this for testing without security plugin? Or it's just temporary workaround as security plugin not ready?

Comment (ohltyler) [reply .github/workflows/e2e-tests-workflow.yml:121]: > > no plugin is able to recognize the security-related settings
> 
> Is this for testing without security plugin? Or it's just temporary workaround as security plugin not ready?

This is for testing without security plugin. Sorry if it's confusing - the workaround is related to adding the `sed` cmd to remove the security settings.

Comment (ohltyler) [reply .github/workflows/e2e-tests-workflow.yml:121]: @ylwu-amzn CI now works with security enabled and disabled: see the latest run [here](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/actions/runs/858565810
)

Comment (ylwu-amzn) [reply .github/workflows/e2e-tests-workflow.yml:121]: Thanks for clarification.

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

File: .github/workflows/e2e-tests-workflow.yml [hunk 2]
```diff
@@ -60,23 +64,22 @@ jobs:
...
         run: |
           cd OpenSearch-Dashboards/plugins/anomaly-detection-dashboards-plugin
-          opensearch_version=$OPENSEARCH_VERSION
           plugin_version=$(node -pe "require('./package.json').version")
-          echo opensearch version: $opensearch_version
+          echo opensearch version: $OPENSEARCH_VERSION
           echo plugin version: $plugin_version
           echo plugin name: $AD_OPENSEARCH_DASHBOARDS_PLUGIN_NAME
-          if docker pull opendistroforelasticsearch/opendistroforelasticsearch:$opensearch_version
+          if docker pull $OPENSEARCH_DOCKER_IMAGE:latest
```
[the remark this answers is not in the document]
Comment (ohltyler) [reply .github/workflows/e2e-tests-workflow.yml:71]: Good question. The idea is that `latest` tag will be updated on a continuous basis and include the latest changes, so we are always testing against the cutting edge. The version associated with `latest` would be the next version that would be released.

Regarding continuous builds and docker tag name conventions: there is still some work to be done to get that standardized and up and running. Leaving as `latest` for now and can update accordingly if it changes.

**Review by spbjss — APPROVED** [review]
Looks good to me.
